### PR TITLE
Fix idempotency isuue about Quota threshold

### PR DESCRIPTION
**Pull request by P-Cao**
# Description
Fix the issue where setting a decimal value for the Quota threshold attribute results in idempotency failure Update mockey dependency

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request

##### RESOURCE OR DATASOURCE NAME

Quota Resource

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by P-Cao** [comment]
run e2e test Quota

**Comment by P-Cao** [comment]
run e2e test Quota

File: powerscale/helper/quota_helper.go [hunk 1]
```diff
@@ -176,3 +178,34 @@ func IsQuotaParamInvalid(plan models.QuotaResource) error {
...
+	priorAttrs := prior.Attributes()
+	for key, priorVal := range priorAttrs {
+		updatedVal, exists := updated.Attributes()[key]
+		if !exists || priorVal.IsNull() || updatedVal.IsNull() {
+			return updated
+		}
+		if numVal1, ok := priorVal.(basetypes.NumberValue); ok {
+
+			if numVal2, ok := updatedVal.(basetypes.NumberValue); ok {
+				if numVal1.ValueBigFloat().String() != numVal2.ValueBigFloat().String() {
```
Comment (baoy1) [inline powerscale/helper/quota_helper.go:202]: Is string comparison preferable to `numVal1.ValueBigFloat().Cmp(numVal2.ValueBigFloat())` ?

File: powerscale/helper/quota_helper.go [hunk 2]
```diff
@@ -176,3 +178,34 @@ func IsQuotaParamInvalid(plan models.QuotaResource) error {
 	}
 	return nil
 }
+
+// UpdateThresholds compares given prior and updated states
+// Due to unknown reason, plugin framework would unmarshal decimal as weird float
+// e.g. advisory_percentage = 10.4 in tf state and advisory_percentage as NumberAttribute in schema
+// would be unmarshalled as something different from types.NumberValue(big.NewFloat())
+// This could cause the problem of
```
Comment (baoy1) [inline powerscale/helper/quota_helper.go:186]: The sentence is incomplete.

File: powerscale/helper/quota_helper.go (hunk 1 above)
Comment (P-Cao) [reply powerscale/helper/quota_helper.go:202]: Actually, that is the problem. 
For example, although the value unmarshalled from `.tfstate` & `types.NumberValue(big.NewFloat(10.4))` are 10.4
image
But they are actually different `BigFloat ` and` .Cmp()` would return false:
image
image

Due to this reason, `terraform plan` would detect the difference even if `.tf` file is not changed.

File: powerscale/helper/quota_helper.go (hunk 2 above)
Comment (P-Cao) [reply powerscale/helper/quota_helper.go]: Fixed

File: powerscale/helper/quota_helper.go (hunk 1 above)
Comment (baoy1) [reply powerscale/helper/quota_helper.go:202]: Thanks for clarifying. It is interesting the two numbers look the same...

**Review by baoy1 — APPROVED** [review]

**Comment by P-Cao** [comment]
run e2e test Quota

**Comment by P-Cao** [comment]
run e2e test Quota

**Comment by P-Cao** [comment]
run e2e test

**Comment by P-Cao** [comment]
run e2e test Quota

**Pull request merged**